plasma_framework/test/src/framework/ExitGameController.test.js

```diff
@@ -0,0 +1,268 @@
+        });
+    });
+
+    describe('processExits', () => {
+        beforeEach(async () => {
+            this.controller.addToken(this.dummyToken);
+            this.dummyExit = {
+                exitProcessor: this.dummyExitGame.address,
+                exitableAt: 1,
+                exitId: 123,
+            };
+        });
+
+        it('rejects when such token has not been added yet', async () => {
+            const fakeNonAddedTokenAddress = (await DummyExitGame.new()).address;
+            await expectRevert(
+                this.controller.processExits(fakeNonAddedTokenAddress, 0, 1),
+                "Such token has not be added to the plasma framework yet"
+            );
+        });
+
+        it('rejects when the exit queue is empty', async () => {
+            await expectRevert(
+                this.controller.processExits(this.dummyToken, 0, 1),
+                "Exit queue is empty"
+            );
+        });
+
+        it('rejects when the "top unique priority" mismatch with the specified one', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            const nonExistingUniqeuePriority = 123;
+            await expectRevert(
+                this.controller.processExits(this.dummyToken, nonExistingUniqeuePriority, 1),
+                "Top unique priority of the queue is not the same as the specified one"
+            );
+        });
+
+        it('processes when the "top unique priority" is set to 0', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            const tx = await this.controller.processExits(this.dummyToken, 0, 1);
+            await expectEvent.inLogs(tx.logs, 'ProcessedExitsNum', {
+                processedNum: new BN(1),
+                token: this.dummyToken
+            });
+        });
+
+        it('processes when the "top unique priority" is set to the exact top one of the queue', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+            const uniquePriority = await this.dummyExitGame.uniqueuPriorityFromEnqueue();
+
+            const tx = await this.controller.processExits(this.dummyToken, uniquePriority, 1);
+            await expectEvent.inLogs(tx.logs, 'ProcessedExitsNum', {
+                processedNum: new BN(1),
+                token: this.dummyToken
+            });
+        });
+
+        it('calls the "processExit" function of the exit processor when processes', async () => {
+            await this.dummyExitGame.enqueue(0, this.dummyToken, this.dummyExit);
+
+            await this.controller.processExits(this.dummyToken, 0, 1);
+
+            // 'exitIdProcessed' would be set when processExit is called in the DummyExitGame
+            const exitIdProcessed = await this.dummyExitGame.exitIdProcessed();
+            expect(exitIdProcessed).to.be.bignumber.equal(new BN(this.dummyExit.exitId));
```

Comment:
I would rather test that by expecting an event that exit has been processed.
I think we are going to deep in implementation details here.

Reply:
Not exactly sure what your recommendation here would be. I make `DummyExitGame` to emit a `ExitFinalizedFromDummyExitGame` event and test that instead of setting exitId variable. Is this what you mean here?

Reply:
My bad, it's ok.